okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -148,6 +148,25 @@ public void onResponse(Call call, Response response) {
         }
     }
 
+    private void cancelCallAndCloseResponseBody(SettableFuture<Response> future) {
+        getDelegate().cancel();
+        executionExecutor.submit(() -> closeResponseBody(future));
```

Comment:
why is a try-finally not sufficient here? why are we putting this onto an executor?

Reply:
in most cases I don't think action is necessary here, could we check if there's a response before putting work on the executor?

Reply:
Getting the future is still blocking until the call completes, in case we failed to cancel. In the case where we were hitting this, it took several seconds for the call to complete before we were able to get the future and close the response body.
If you have a better way of doing this, I'm happy to change :)

Reply:
@gmaretic just revisiting this - couldn't we check the `future.isDone()` method to find out of the call is finished (in a non-blocking way) and avoid submitting stuff to the executor?

I'm thinking that this would be beneficial because the `catch (ExecutionException e)` branch seems a lot more common than the `catch (InterruptedException e)` one.

Reply:
Wouldn't it be simplest to just add a listener to the future using `future.transform(...)`? Then we don't need to check whether the future is done or submit anything to an executor.